using System.Drawing; public class Donkey {   public Image Image { get;set; } }

docmaster2 opened the pull request:
<!-- Please provide a brief summary of your changes: -->

<!-- Please ensure you’ve done all of these things (if applicable). -->
<!-- You can replace the `[ ]` with `[x]` to mark each task as done. -->

- [x] I’ve added tests to confirm my change works.
- [x] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

docmaster246 reviewed (commented):
@docmaster2 
![ddrup5s-b98539bc-6331-4fdd-9db1-8e8336435516](https://user-images.githubusercontent.com/65420864/82125907-bc288880-97b1-11ea-86f2-1d0713aaa184.png)
DA MURO в отличной форме

docmaster246 reviewed (changes requested):
// Updated test case with a Link to a different address
it('renders correctly', () => {
  const tree = renderer
    .create(<Link page="http://www.instagram.com">Instagram</Link>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
![водопад(2)](https://user-images.githubusercontent.com/65420864/82145097-1d536900-9851-11ea-9968-c3c04da154e8.png)

sosukesuzuki committed:
Fix formatting for AssignmentExpression with ClassExpression (#9741)

* Fix formatting for asignment with superClass

* Add tests

* Changelog

* For only AssignmentExpression

* Remove unnecesary ifBreak

fisker committed:
Refactor `ConditionalExpression` and `TSConditionalType` print (#9753)

dependabot[bot] committed:
Build(deps-dev): Bump @glimmer/reference from 0.66.0 to 0.66.1 (#9752)

fisker committed:
Move `Declare*` node to `print/flow.js` (#9757)

fisker committed:
Move some "TS*" node to `print/typescript.js` (#9758)

dependabot[bot] committed:
Build(deps): Bump @glimmer/syntax from 0.66.0 to 0.66.1 (#9747)

Bumps [@glimmer/syntax](https://github.com/glimmerjs/glimmer-vm) from 0.66.0 to 0.66.1.
- [Release notes](https://github.com/glimmerjs/glimmer-vm/releases)
- [Changelog](https://github.com/glimmerjs/glimmer-vm/blob/master/CHANGELOG.md)
- [Commits](https://github.com/glimmerjs/glimmer-vm/compare/v0.66.0...v0.66.1)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps): Bump @babel/parser from 7.12.5 to 7.12.7 (#9748)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: fisker <lionkay@gmail.com>

kangax committed:
[Glimmer] Preserve tilde comments (#9082)

Co-authored-by: fisker <lionkay@gmail.com>

dependabot[bot] committed:
Build(deps-dev): Bump @babel/core from 7.12.7 to 7.12.8 (#9765)

fisker committed:
Extract some type annotation to `flow.js` and `typescript.js` (#9761)

fisker committed:
Test: Fix AST_COMPARE test (#9768)

dependabot[bot] committed:
Build(deps-dev): Bump @typescript-eslint/types from 4.8.1 to 4.8.2 (#9766)

dependabot[bot] committed:
Build(deps): Bump @typescript-eslint/typescript-estree (#9763)

fisker committed:
Remove `JSXText` spacial case in comment attaching (#9759)

dependabot[bot] committed:
Build(deps-dev): Bump @glimmer/reference from 0.66.1 to 0.67.0 (#9764)

fisker committed:
Fix yarn.lock

dependabot[bot] committed:
Build(deps): Bump @glimmer/syntax from 0.66.1 to 0.67.0 (#9762)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: fisker <lionkay@gmail.com>

fisker committed:
Rename a variable (#9769)

fisker committed:
Add `semi` to `DeclareExportAllDeclaration` (#9767)

fisker committed:
`printDocToString`: Don't check existance before replacing newLine (#9775)

fisker committed:
JS: Add `hasComment` and `getComments` (#9742)

sosukesuzuki committed:
Fix changelog to old format

fisker committed:
SCSS/LESS: Fix comment inside parens (#9710)

fisker committed:
Fix inconsistent language comment detection (#9743)

fisker committed:
Fix `lint:changelog`

thorn0 committed:
Update option philosophy: option requests aren’t accepted anymore (#8540)

* Update option philosophy: option requests aren’t accepted anymore

* Update docs/option-philosophy.md

Co-authored-by: Lucas Azzola <lucas.azzola@iress.com>

* Update option-philosophy.md

Co-authored-by: Lucas Azzola <lucas.azzola@iress.com>

thorn0 committed:
copy option philosophy to stable docs

dependabot[bot] committed:
Build(deps-dev): Bump babel-loader from 8.2.1 to 8.2.2 in /website (#9778)

dependabot[bot] committed:
Build(deps-dev): Bump webpack from 5.6.0 to 5.8.0 in /website (#9779)

dependabot[bot] committed:
Build(deps-dev): Bump babel-loader from 8.2.1 to 8.2.2 (#9781)

dependabot[bot] committed:
Build(deps-dev): Bump webpack from 5.6.0 to 5.8.0 (#9780)

oebrab committed:
Add a link to the "Example" part (#9782)

sosukesuzuki committed:
Release 2.2.1

sosukesuzuki committed:
Git blame ignore 2.2.1

sosukesuzuki committed:
Bump Prettier dependency to 2.2.1

sosukesuzuki committed:
Update dependents count

sosukesuzuki committed:
Merge branch 'patch-release'

sosukesuzuki committed:
Clean changelog_unreleased for patch-release

sosukesuzuki committed:
Clean changelog_unreleased for patch-release

dependabot[bot] committed:
Build(deps-dev): Bump @babel/core from 7.12.8 to 7.12.9 (#9774)

Bumps [@babel/core](https://github.com/babel/babel/tree/HEAD/packages/babel-core) from 7.12.8 to 7.12.9.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.12.9/packages/babel-core)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

fisker committed:
Treat `.prettierrc` as yaml (#8105)

fisker committed:
Exclude node at `rangeStart` (#9704)

fisker committed:
JS: Fix multiple-comments-on-same-line format (#9672)

Co-authored-by: Georgii Dolzhykov <thorn.mailbox@gmail.com>

dependabot[bot] committed:
Build(deps): Bump codemirror-graphql from 0.12.3 to 0.12.4 in /website (#9792)

Bumps [codemirror-graphql](https://github.com/graphql/graphiql) from 0.12.3 to 0.12.4.
- [Release notes](https://github.com/graphql/graphiql/releases)
- [Changelog](https://github.com/graphql/graphiql/blob/main/CHANGELOG.md)
- [Commits](https://github.com/graphql/graphiql/compare/codemirror-graphql@0.12.3...codemirror-graphql@0.12.4)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump webpack from 5.8.0 to 5.9.0 in /website (#9793)

dependabot[bot] committed:
Build(deps-dev): Bump cspell from 4.2.2 to 4.2.3 (#9796)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump rollup from 2.33.3 to 2.34.0 (#9795)

Bumps [rollup](https://github.com/rollup/rollup) from 2.33.3 to 2.34.0.
- [Release notes](https://github.com/rollup/rollup/releases)
- [Changelog](https://github.com/rollup/rollup/blob/master/CHANGELOG.md)
- [Commits](https://github.com/rollup/rollup/compare/v2.33.3...v2.34.0)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump webpack from 5.8.0 to 5.9.0 (#9794)

Bumps [webpack](https://github.com/webpack/webpack) from 5.8.0 to 5.9.0.
- [Release notes](https://github.com/webpack/webpack/releases)
- [Commits](https://github.com/webpack/webpack/compare/v5.8.0...v5.9.0)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

fisker committed:
Fix empty front matter formatting (#9791)

fisker committed:
Refactor: Stop mutating AST during `getSortedChildNodes` (#9797)

dependabot[bot] committed:
Build(deps-dev): Bump cspell from 4.2.3 to 4.2.5 (#9803)

Bumps [cspell](https://github.com/streetsidesoftware/cspell) from 4.2.3 to 4.2.5.
- [Release notes](https://github.com/streetsidesoftware/cspell/releases)
- [Changelog](https://github.com/streetsidesoftware/cspell/blob/master/CHANGELOG.md)
- [Commits](https://github.com/streetsidesoftware/cspell/compare/cspell@4.2.3...cspell@4.2.5)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump @typescript-eslint/types from 4.8.2 to 4.9.0 (#9802)

Bumps [@typescript-eslint/types](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/visitor-keys) from 4.8.2 to 4.9.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/visitor-keys/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v4.9.0/packages/visitor-keys)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump @rollup/plugin-babel from 5.2.1 to 5.2.2 (#9805)

Bumps [@rollup/plugin-babel](https://github.com/rollup/plugins) from 5.2.1 to 5.2.2.
- [Release notes](https://github.com/rollup/plugins/releases)
- [Commits](https://github.com/rollup/plugins/compare/babel-v5.2.1...babel-v5.2.2)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

fisker committed:
Build: Fix rollup warning (#9808)

sosukesuzuki committed:
Refactor: Make Cache modern in build scripts (#9809)

* Make cache class

* equal -> strictEqual

* Use strict mode assert module

dependabot[bot] committed:
Build(deps-dev): Bump cross-env from 7.0.2 to 7.0.3 (#9814)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump @rollup/plugin-node-resolve from 10.0.0 to 11.0.0 (#9804)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: fisker <lionkay@gmail.com>

thorn0 committed:
Fix an oversight from #9709

dependabot[bot] committed:
Build(deps): Bump semver from 7.3.2 to 7.3.4 (#9813)

Bumps [semver](https://github.com/npm/node-semver) from 7.3.2 to 7.3.4.
- [Release notes](https://github.com/npm/node-semver/releases)
- [Changelog](https://github.com/npm/node-semver/blob/master/CHANGELOG.md)
- [Commits](https://github.com/npm/node-semver/compare/v7.3.2...v7.3.4)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps): Bump @typescript-eslint/typescript-estree from 4.8.2 to 4.9.0 (#9806)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump @glimmer/reference from 0.67.0 to 0.68.0 (#9821)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump rollup from 2.34.0 to 2.34.1 (#9823)

dependabot[bot] committed:
Build(deps-dev): Bump execa from 4.1.0 to 5.0.0 (#9820)

dependabot[bot] committed:
Build(deps): Bump @glimmer/syntax from 0.67.0 to 0.68.0 (#9822)

Bumps [@glimmer/syntax](https://github.com/glimmerjs/glimmer-vm) from 0.67.0 to 0.68.0.
- [Release notes](https://github.com/glimmerjs/glimmer-vm/releases)
- [Changelog](https://github.com/glimmerjs/glimmer-vm/blob/master/CHANGELOG.md)
- [Commits](https://github.com/glimmerjs/glimmer-vm/compare/v0.67.0...v0.68.0)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

fisker committed:
Remove dead code (#9824)

fisker committed:
Fix `stripTrailingHardline` function (#9736)

Co-authored-by: Georgii Dolzhykov <thorn.mailbox@gmail.com>

fisker committed:
Update `@angular/compiler` and `angular-estree-parser` (#9816)

dependabot[bot] committed:
Build(deps-dev): Bump eslint from 7.14.0 to 7.15.0 (#9836)

dependabot[bot] committed:
Build(deps-dev): Bump @glimmer/reference from 0.68.0 to 0.68.1 (#9833)

dependabot[bot] committed:
Build(deps-dev): Bump webpack from 5.9.0 to 5.10.0 in /website (#9832)

dependabot[bot] committed:
Build(deps): Bump @glimmer/syntax from 0.68.0 to 0.68.1 (#9835)

dependabot[bot] committed:
Build(deps-dev): Bump webpack from 5.9.0 to 5.10.0 (#9834)

dependabot[bot] committed:
Build(deps-dev): Bump eslint-config-prettier from 6.15.0 to 7.0.0 (#9838)

dependabot[bot] committed:
Build(deps): Bump espree from 7.3.0 to 7.3.1 (#9837)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump rollup from 2.34.1 to 2.34.2 (#9831)

sosukesuzuki committed:
Print null argument list item (#9787)

* Print null argument list item

* Add changelog

* Add tests for comments

sosukesuzuki committed:
Extract files and rename util -> core (#9826)

WIP

Extract create-minimist-options

Remove warning

Fix

Extract format.js

Fix

Remove

fix

Fix

Extract files for cli and rename util -> core

Update tsconfig

dependabot[bot] committed:
Build(deps-dev): Bump js-yaml from 3.14.0 to 3.14.1 in /website (#9843)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps): Bump @typescript-eslint/typescript-estree from 4.9.0 to 4.9.1 (#9842)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump @typescript-eslint/types from 4.9.0 to 4.9.1 (#9840)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps): Bump codemirror-graphql from 0.12.4 to 0.13.0 in /website (#9851)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps): Bump flow-parser from 0.138.0 to 0.139.0 (#9841)

Bumps [flow-parser](https://github.com/facebook/flow) from 0.138.0 to 0.139.0.
- [Release notes](https://github.com/facebook/flow/releases)
- [Changelog](https://github.com/facebook/flow/blob/master/Changelog.md)
- [Commits](https://github.com/facebook/flow/compare/v0.138.0...v0.139.0)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump @babel/core from 7.12.9 to 7.12.10 (#9859)

dependabot[bot] committed:
Build(deps): Bump @angular/compiler from 11.0.3 to 11.0.4 (#9864)

dependabot[bot] committed:
Build(deps-dev): Bump @babel/types from 7.12.7 to 7.12.10 (#9858)

dependabot[bot] committed:
Build(deps-dev): Bump @babel/preset-env from 7.12.7 to 7.12.10 (#9863)

dependabot[bot] committed:
Build(deps-dev): Bump @glimmer/reference from 0.68.1 to 0.69.0 (#9860)

dependabot[bot] committed:
Build(deps): Bump @glimmer/syntax from 0.68.1 to 0.69.0 (#9862)

dependabot[bot] committed:
Build(deps-dev): Bump @babel/preset-react in /website (#9861)

Bumps [@babel/preset-react](https://github.com/babel/babel/tree/HEAD/packages/babel-preset-react) from 7.12.7 to 7.12.10.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.12.10/packages/babel-preset-react)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

fisker committed:
HTML: Remove use of `normalizeParts` in "text" print (#9846)

dependabot[bot] committed:
Build(deps): Bump actions/setup-node from v2.1.2 to v2.1.3 (#9870)

dependabot[bot] committed:
Build(deps-dev): Bump @glimmer/reference from 0.69.0 to 0.69.3 (#9868)

dependabot[bot] committed:
Build(deps): Bump @glimmer/syntax from 0.69.0 to 0.69.3 (#9869)

dependabot[bot] committed:
Build(deps): Bump @babel/parser from 7.12.7 to 7.12.10 (#9865)

Bumps [@babel/parser](https://github.com/babel/babel/tree/HEAD/packages/babel-parser) from 7.12.7 to 7.12.10.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.12.10/packages/babel-parser)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

kddnewton committed:
Ruby blog post (#9854)

kddnewton committed:
Correct name for Ruby blog post (#9875)

fisker committed:
Fix link to babel website (#9856)

Co-authored-by: Sosuke Suzuki <aosukeke@gmail.com>

dependabot[bot] committed:
Build(deps): Bump flow-parser from 0.139.0 to 0.140.0 (#9867)

Bumps [flow-parser](https://github.com/facebook/flow) from 0.139.0 to 0.140.0.
- [Release notes](https://github.com/facebook/flow/releases)
- [Changelog](https://github.com/facebook/flow/blob/master/Changelog.md)
- [Commits](https://github.com/facebook/flow/compare/v0.139.0...v0.140.0)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

fisker committed:
doc: improve examples in browser (#9811)

fisker committed:
Use `isBlockComment`/`isLineComment` in `canAttachComment` (#9849)

dependabot[bot] committed:
Build(deps-dev): Bump webpack from 5.10.0 to 5.10.1 in /website (#9886)

Bumps [webpack](https://github.com/webpack/webpack) from 5.10.0 to 5.10.1.
- [Release notes](https://github.com/webpack/webpack/releases)
- [Commits](https://github.com/webpack/webpack/compare/v5.10.0...v5.10.1)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump webpack from 5.10.0 to 5.10.1 (#9889)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps): Bump linguist-languages from 7.12.1 to 7.12.2 (#9887)

Bumps [linguist-languages](https://github.com/ikatyang/linguist-languages) from 7.12.1 to 7.12.2.
- [Release notes](https://github.com/ikatyang/linguist-languages/releases)
- [Commits](https://github.com/ikatyang/linguist-languages/compare/v7.12.1...v7.12.2)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps): Bump typescript from 4.1.2 to 4.1.3 (#9888)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

fisker committed:
TypeScript: Fix missing comments in `MethodDefinition` (#9872)

fisker committed:
Consistent `Jsx` case in variables (#9871)

fisker committed:
Fix inconsistent directive print (#9850)

fisker committed:
Fix unstable JSX formatting with U+3000 (#9866)

dependabot[bot] committed:
Build(deps-dev): Bump rollup from 2.34.2 to 2.35.1 (#9891)

dependabot[bot] committed:
Build(deps): Bump @glimmer/syntax from 0.69.3 to 0.70.0 (#9890)

dependabot[bot] committed:
Build(deps-dev): Bump @rollup/plugin-node-resolve from 11.0.0 to 11.0.1 (#9895)

fisker committed:
Refactor: Improve JavaScript parser readability (#9783)

dependabot[bot] committed:
Build(deps-dev): Bump @typescript-eslint/types from 4.9.1 to 4.10.0 (#9894)

dependabot[bot] committed:
Build(deps): Bump @typescript-eslint/typescript-estree (#9892)

dependabot[bot] committed:
Build(deps-dev): Bump @glimmer/reference from 0.69.3 to 0.70.0 (#9893)

fisker committed:
Fix `yarn.lock`

fisker committed:
Refactor: Extract `printBody` (#9898)

sosukesuzuki committed:
Prints bundle sizes in build script log (#9900)

* Prints bundle size with flags

* Add "gzipped" to cspell.json

* Print esm bundle size

* Remove gzipped size from log

fisker committed:
Simplify `escapeTemplateCharacters` (#9848)

fisker committed:
Refactor: Simplify JS parsers (#9897)

fisker committed:
Refactor: Remove `composeLoc` function (#9896)

dependabot[bot] committed:
Build(deps): Bump @glimmer/syntax from 0.70.0 to 0.71.0 (#9908)

dependabot[bot] committed:
Build(deps-dev): Bump webpack from 5.10.1 to 5.10.3 in /website (#9903)

dependabot[bot] committed:
Build(deps-dev): Bump @babel/preset-env from 7.12.10 to 7.12.11 (#9911)

dependabot[bot] committed:
Build(deps-dev): Bump webpack from 5.10.1 to 5.10.3 (#9909)

sosukesuzuki committed:
Fix params mistaking at #9900 (#9912)

dependabot[bot] committed:
Build(deps): Bump @babel/code-frame from 7.10.4 to 7.12.11 (#9906)

dependabot[bot] committed:
Build(deps-dev): Bump @babel/types from 7.12.10 to 7.12.11 (#9910)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump @glimmer/reference from 0.70.0 to 0.71.0 (#9905)

dependabot[bot] committed:
Build(deps): Bump @babel/parser from 7.12.10 to 7.12.11 (#9904)

Bumps [@babel/parser](https://github.com/babel/babel/tree/HEAD/packages/babel-parser) from 7.12.10 to 7.12.11.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.12.11/packages/babel-parser)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

fisker committed:
Replace `normalizeParts` with safer `cleanDoc` (#9844)

fisker committed:
Refactor(html): use `concat` directly (#9845)

fisker committed:
DRY Block and Program print (#9899)

fisker committed:
Refactor yaml (#9629)

fisker committed:
Use `index` instead of `.getName()` in `FastPath#each()` (#9914)

fisker committed:
Fix `comments.js` import style (#9917)

fisker committed:
Move `isLastStatement` to `print/statement.js` (#9916)

dependabot[bot] committed:
Build(deps): Bump actions/setup-node from v2.1.3 to v2.1.4 (#9924)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps): Bump outdent from 0.7.1 to 0.8.0 (#9923)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps): Bump @glimmer/syntax from 0.71.0 to 0.72.0 (#9922)

Bumps [@glimmer/syntax](https://github.com/glimmerjs/glimmer-vm) from 0.71.0 to 0.72.0.
- [Release notes](https://github.com/glimmerjs/glimmer-vm/releases)
- [Changelog](https://github.com/glimmerjs/glimmer-vm/blob/master/CHANGELOG.md)
- [Commits](https://github.com/glimmerjs/glimmer-vm/compare/v0.71.0...v0.72.0)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps): Bump @angular/compiler from 11.0.4 to 11.0.5 (#9921)

Bumps [@angular/compiler](https://github.com/angular/angular/tree/HEAD/packages/compiler) from 11.0.4 to 11.0.5.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/HEAD/packages/compiler)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

fisker committed:
Enable `arrow-body-style` rule (#9918)

fisker committed:
Add `isNonEmptyArray` (#9902)

fisker committed:
Send array to `FastPath#{each,map}` callback (#9901)

dependabot[bot] committed:
Build(deps-dev): Bump @glimmer/reference from 0.71.0 to 0.72.0 (#9919)

fisker committed:
Add JS version `isNextLineEmpty` (#9925)

dependabot[bot] committed:
Build(deps-dev): Bump eslint-plugin-unicorn from 23.0.0 to 24.0.0 (#9931)

Bumps [eslint-plugin-unicorn](https://github.com/sindresorhus/eslint-plugin-unicorn) from 23.0.0 to 24.0.0.
- [Release notes](https://github.com/sindresorhus/eslint-plugin-unicorn/releases)
- [Commits](https://github.com/sindresorhus/eslint-plugin-unicorn/compare/v23.0.0...v24.0.0)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump webpack from 5.10.3 to 5.11.0 (#9932)

dependabot[bot] committed:
Build(deps-dev): Bump webpack from 5.10.3 to 5.11.0 in /website (#9930)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps): Bump codecov/codecov-action from v1.0.15 to v1.1.0 (#9934)

Bumps [codecov/codecov-action](https://github.com/codecov/codecov-action) from v1.0.15 to v1.1.0.
- [Release notes](https://github.com/codecov/codecov-action/releases)
- [Changelog](https://github.com/codecov/codecov-action/blob/master/CHANGELOG.md)
- [Commits](https://github.com/codecov/codecov-action/compare/v1.0.15...7de43a7373de21874ae196a78f8eb633fcf7f0c4)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

fisker committed:
Refactor `graphql` parser to use `tryCombinations` (#9928)

fisker committed:
Refactor: Move `jsx` related function to `jsx.js` (#9927)

dependabot[bot] committed:
Build(deps): Bump preactjs/compressed-size-action from v2.0.1 to 2.1.0 (#9942)

Bumps [preactjs/compressed-size-action](https://github.com/preactjs/compressed-size-action) from v2.0.1 to 2.1.0.
- [Release notes](https://github.com/preactjs/compressed-size-action/releases)
- [Commits](https://github.com/preactjs/compressed-size-action/compare/v2.0.1...7d87f60a6b0c7d193b8183ce859ed00b356ea92f)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump eslint-config-prettier from 7.0.0 to 7.1.0 (#9940)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump @glimmer/reference from 0.72.0 to 0.73.0 (#9941)

Bumps @glimmer/reference from 0.72.0 to 0.73.0.

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps): Bump flow-parser from 0.140.0 to 0.141.0 (#9938)

Bumps [flow-parser](https://github.com/facebook/flow) from 0.140.0 to 0.141.0.
- [Release notes](https://github.com/facebook/flow/releases)
- [Changelog](https://github.com/facebook/flow/blob/master/Changelog.md)
- [Commits](https://github.com/facebook/flow/compare/v0.140.0...v0.141.0)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dcyriller committed:
Handlebars: Add tests for `ignore` and `strict` modes (1/3) (#9883)

dependabot[bot] committed:
Build(deps-dev): Bump eslint from 7.15.0 to 7.16.0 (#9939)

fisker committed:
Enable `unicorn/prefer-number-properties` (#9935)

fisker committed:
Use `printBlock` to print `ClassBody` (#9915)

dependabot[bot] committed:
Build(deps): Bump @glimmer/syntax from 0.72.0 to 0.73.0 (#9937)

Bumps [@glimmer/syntax](https://github.com/glimmerjs/glimmer-vm) from 0.72.0 to 0.73.0.
- [Release notes](https://github.com/glimmerjs/glimmer-vm/releases)
- [Changelog](https://github.com/glimmerjs/glimmer-vm/blob/master/CHANGELOG.md)
- [Commits](https://github.com/glimmerjs/glimmer-vm/compare/v0.72.0...v0.73.0)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump @glimmer/reference from 0.73.0 to 0.73.1 (#9945)

Bumps @glimmer/reference from 0.73.0 to 0.73.1.

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump @typescript-eslint/types from 4.10.0 to 4.11.0 (#9947)

Bumps [@typescript-eslint/types](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/visitor-keys) from 4.10.0 to 4.11.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/visitor-keys/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v4.11.0/packages/visitor-keys)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps): Bump @glimmer/syntax from 0.73.0 to 0.73.1 (#9944)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps): Bump @typescript-eslint/typescript-estree from 4.10.0 to 4.11.0 (#9946)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps): Bump codecov/codecov-action from v1.1.0 to v1.1.1 (#9948)

Bumps [codecov/codecov-action](https://github.com/codecov/codecov-action) from v1.1.0 to v1.1.1.
- [Release notes](https://github.com/codecov/codecov-action/releases)
- [Changelog](https://github.com/codecov/codecov-action/blob/master/CHANGELOG.md)
- [Commits](https://github.com/codecov/codecov-action/compare/v1.1.0...1fc7722ded4708880a5aea49f2bfafb9336f0c8d)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump builtin-modules from 3.1.0 to 3.2.0 (#9950)

Bumps [builtin-modules](https://github.com/sindresorhus/builtin-modules) from 3.1.0 to 3.2.0.
- [Release notes](https://github.com/sindresorhus/builtin-modules/releases)
- [Commits](https://github.com/sindresorhus/builtin-modules/compare/v3.1.0...v3.2.0)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

fisker committed:
Test internal eslint rules (#9929)

dependabot[bot] committed:
Build(deps-dev): Bump @babel/types from 7.12.11 to 7.12.12 (#9957)

Bumps [@babel/types](https://github.com/babel/babel/tree/HEAD/packages/babel-types) from 7.12.11 to 7.12.12.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.12.12/packages/babel-types)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump pretty-bytes from 5.4.1 to 5.5.0 (#9956)

Bumps [pretty-bytes](https://github.com/sindresorhus/pretty-bytes) from 5.4.1 to 5.5.0.
- [Release notes](https://github.com/sindresorhus/pretty-bytes/releases)
- [Commits](https://github.com/sindresorhus/pretty-bytes/compare/v5.4.1...v5.5.0)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

vjeux committed:
Fix absolute path in custom CSS url() calls (#9966)

At Facebook we have some custom pre-processor `-fb-url(/abs/path/)` that gets messed up as `-fb-url(/ abs/path)`  with the current version of prettier.

The parser has hardcoded logic to only parse as path things that use `url()` but not `-fb-url()` and it parses it as `division /` and `abs/path/`. Because we put a space after division, the bug is introduced.

I don't really know the impact of changing the parser, but on the printer side, it is straightforward to not add a space after `/` if it's the first element in a group. I can't think of any reason why you could write `(/ something` in CSS so I think it's a safe change.

dependabot[bot] committed:
Build(deps): Bump @typescript-eslint/typescript-estree (#9968)

Bumps [@typescript-eslint/typescript-estree](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/typescript-estree) from 4.11.0 to 4.11.1.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/typescript-estree/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v4.11.1/packages/typescript-estree)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump webpack from 5.11.0 to 5.11.1 in /website (#9970)

Bumps [webpack](https://github.com/webpack/webpack) from 5.11.0 to 5.11.1.
- [Release notes](https://github.com/webpack/webpack/releases)
- [Commits](https://github.com/webpack/webpack/compare/v5.11.0...v5.11.1)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps): Bump codemirror-graphql from 0.13.0 to 0.13.1 in /website (#9971)

Bumps [codemirror-graphql](https://github.com/graphql/graphiql) from 0.13.0 to 0.13.1.
- [Release notes](https://github.com/graphql/graphiql/releases)
- [Changelog](https://github.com/graphql/graphiql/blob/main/CHANGELOG.md)
- [Commits](https://github.com/graphql/graphiql/compare/codemirror-graphql@0.13.0...codemirror-graphql@0.13.1)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump eslint-plugin-unicorn from 24.0.0 to 25.0.0 (#9969)

Bumps [eslint-plugin-unicorn](https://github.com/sindresorhus/eslint-plugin-unicorn) from 24.0.0 to 25.0.0.
- [Release notes](https://github.com/sindresorhus/eslint-plugin-unicorn/releases)
- [Commits](https://github.com/sindresorhus/eslint-plugin-unicorn/compare/v24.0.0...v25.0.0)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump webpack from 5.11.0 to 5.11.1 (#9974)

Bumps [webpack](https://github.com/webpack/webpack) from 5.11.0 to 5.11.1.
- [Release notes](https://github.com/webpack/webpack/releases)
- [Commits](https://github.com/webpack/webpack/compare/v5.11.0...v5.11.1)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump webpack-cli from 4.2.0 to 4.3.0 in /website (#9962)

Bumps [webpack-cli](https://github.com/webpack/webpack-cli) from 4.2.0 to 4.3.0.
- [Release notes](https://github.com/webpack/webpack-cli/releases)
- [Changelog](https://github.com/webpack/webpack-cli/blob/master/CHANGELOG.md)
- [Commits](https://github.com/webpack/webpack-cli/compare/webpack-cli@4.2.0...webpack-cli@4.3.0)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump eslint-plugin-react from 7.21.5 to 7.22.0 (#9976)

Bumps [eslint-plugin-react](https://github.com/yannickcr/eslint-plugin-react) from 7.21.5 to 7.22.0.
- [Release notes](https://github.com/yannickcr/eslint-plugin-react/releases)
- [Changelog](https://github.com/yannickcr/eslint-plugin-react/blob/master/CHANGELOG.md)
- [Commits](https://github.com/yannickcr/eslint-plugin-react/compare/v7.21.5...v7.22.0)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump @typescript-eslint/types from 4.11.0 to 4.11.1 (#9973)

Bumps [@typescript-eslint/types](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/visitor-keys) from 4.11.0 to 4.11.1.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/visitor-keys/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v4.11.1/packages/visitor-keys)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump eslint-plugin-unicorn from 25.0.0 to 25.0.1 (#9983)

Bumps [eslint-plugin-unicorn](https://github.com/sindresorhus/eslint-plugin-unicorn) from 25.0.0 to 25.0.1.
- [Release notes](https://github.com/sindresorhus/eslint-plugin-unicorn/releases)
- [Commits](https://github.com/sindresorhus/eslint-plugin-unicorn/compare/v25.0.0...v25.0.1)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

fisker committed:
Add test to close issue 9102 (#9982)

sosukesuzuki committed:
Refactor CLI context with ES2015 class (#9964)

* Remove createContext

* Remove initContext

* Remove pushContextPlugins

* Remove popContextPlugins

* Move JSDoc comments

* Remove updateContextArgv

* Remove normalizeContextArgv

* Remove updateContextOptions

* Fix typedef

* Change exports format

* Add comment for initContext

fisker committed:
Add `Doc[]` as `Doc` (#9733)

Co-authored-by: Georgii Dolzhykov <thorn.mailbox@gmail.com>

fisker committed:
Enable `unicorn/explicit-length-check` (#9975)

fisker committed:
Simplify `printBody` (#9926)

dependabot[bot] committed:
Build(deps): Bump codemirror-graphql from 0.13.1 to 0.14.0 in /website (#9997)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

fisker committed:
Fix using `concat()` on single `Doc` (#9994)

Co-authored-by: Sosuke Suzuki <aosukeke@gmail.com>

dependabot[bot] committed:
Build(deps-dev): Bump eslint from 7.16.0 to 7.17.0 (#9995)

Bumps [eslint](https://github.com/eslint/eslint) from 7.16.0 to 7.17.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/master/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v7.16.0...v7.17.0)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump webpack-cli from 4.3.0 to 4.3.1 in /website (#9989)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps): Bump @typescript-eslint/typescript-estree from 4.11.1 to 4.12.0 (#10003)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps): Bump codecov/codecov-action from v1.1.1 to v1.2.0 (#10004)

Bumps [codecov/codecov-action](https://github.com/codecov/codecov-action) from v1.1.1 to v1.2.0.
- [Release notes](https://github.com/codecov/codecov-action/releases)
- [Changelog](https://github.com/codecov/codecov-action/blob/master/CHANGELOG.md)
- [Commits](https://github.com/codecov/codecov-action/compare/v1.1.1...a92c414703a4bba586f6df7fcc885c9d0bdff772)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump @typescript-eslint/types from 4.11.1 to 4.12.0 (#10002)

dependabot[bot] committed:
Build(deps-dev): Bump rollup from 2.35.1 to 2.36.0 (#10010)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps): Bump codecov/codecov-action from v1.2.0 to v1.2.1 (#10011)

Bumps [codecov/codecov-action](https://github.com/codecov/codecov-action) from v1.2.0 to v1.2.1.
- [Release notes](https://github.com/codecov/codecov-action/releases)
- [Changelog](https://github.com/codecov/codecov-action/blob/master/CHANGELOG.md)
- [Commits](https://github.com/codecov/codecov-action/compare/v1.2.0...e156083f13aff6830c92fc5faa23505779fbf649)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps): Bump @angular/compiler from 11.0.5 to 11.0.6 (#10017)

Bumps [@angular/compiler](https://github.com/angular/angular/tree/HEAD/packages/compiler) from 11.0.5 to 11.0.6.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/11.0.6/packages/compiler)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps): Bump flow-parser from 0.141.0 to 0.142.0 (#10016)

Bumps [flow-parser](https://github.com/facebook/flow) from 0.141.0 to 0.142.0.
- [Release notes](https://github.com/facebook/flow/releases)
- [Changelog](https://github.com/facebook/flow/blob/master/Changelog.md)
- [Commits](https://github.com/facebook/flow/compare/v0.141.0...v0.142.0)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps): Bump globby from 11.0.1 to 11.0.2 (#10015)

Bumps [globby](https://github.com/sindresorhus/globby) from 11.0.1 to 11.0.2.
- [Release notes](https://github.com/sindresorhus/globby/releases)
- [Commits](https://github.com/sindresorhus/globby/compare/v11.0.1...v11.0.2)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump rollup from 2.36.0 to 2.36.1 (#10014)

Bumps [rollup](https://github.com/rollup/rollup) from 2.36.0 to 2.36.1.
- [Release notes](https://github.com/rollup/rollup/releases)
- [Changelog](https://github.com/rollup/rollup/blob/master/CHANGELOG.md)
- [Commits](https://github.com/rollup/rollup/compare/v2.36.0...v2.36.1)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

sosukesuzuki committed:
Use fs.promises in build script (#10025)

dependabot[bot] committed:
Build(deps): Bump codemirror-graphql from 0.14.0 to 0.15.2 in /website (#10020)

Bumps [codemirror-graphql](https://github.com/graphql/graphiql) from 0.14.0 to 0.15.2.
- [Release notes](https://github.com/graphql/graphiql/releases)
- [Changelog](https://github.com/graphql/graphiql/blob/main/CHANGELOG.md)
- [Commits](https://github.com/graphql/graphiql/compare/codemirror-graphql@0.14.0...codemirror-graphql@0.15.2)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps): Bump cjk-regex from 2.0.0 to 2.0.1 (#10021)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: sosukesuzuki <aosukeke@gmail.com>

dependabot[bot] committed:
Build(deps): Bump @angular/compiler from 11.0.6 to 11.0.7 (#10022)

Bumps [@angular/compiler](https://github.com/angular/angular/tree/HEAD/packages/compiler) from 11.0.6 to 11.0.7.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/11.0.7/packages/compiler)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

fisker committed:
Forbid use `concat()` in `concat()` (#9998)

sosukesuzuki committed:
Check cache only when purge-cache is off (#10031)

Co-authored-by: fisker Cheung <lionkay@gmail.com>

sosukesuzuki committed:
Move checkCache to Cache class (#10029)

* Move checkCache to Cache class

* Address reviews

fisker committed:
Refactor: Use `Doc[]` instead of `concat()` (#9993)

dependabot[bot] committed:
Build(deps-dev): Bump webpack from 5.11.1 to 5.12.3 in /website (#10034)

Bumps [webpack](https://github.com/webpack/webpack) from 5.11.1 to 5.12.3.
- [Release notes](https://github.com/webpack/webpack/releases)
- [Commits](https://github.com/webpack/webpack/compare/v5.11.1...v5.12.3)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump webpack from 5.11.1 to 5.12.3 (#10035)

Bumps [webpack](https://github.com/webpack/webpack) from 5.11.1 to 5.12.3.
- [Release notes](https://github.com/webpack/webpack/releases)
- [Commits](https://github.com/webpack/webpack/compare/v5.11.1...v5.12.3)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump terser-webpack-plugin from 5.0.3 to 5.1.1 (#10033)

dependabot[bot] committed:
Build(deps-dev): Bump @types/estree from 0.0.45 to 0.0.46 (#10043)

Bumps [@types/estree](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/estree) from 0.0.45 to 0.0.46.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/estree)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump webpack from 5.12.3 to 5.13.0 in /website (#10037)

Bumps [webpack](https://github.com/webpack/webpack) from 5.12.3 to 5.13.0.
- [Release notes](https://github.com/webpack/webpack/releases)
- [Commits](https://github.com/webpack/webpack/compare/v5.12.3...v5.13.0)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps): Bump @angular/compiler from 11.0.7 to 11.0.8 (#10041)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump eslint-plugin-unicorn from 25.0.1 to 26.0.0 (#10042)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps): Bump @typescript-eslint/typescript-estree from 4.12.0 to 4.13.0 (#10039)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump @typescript-eslint/types from 4.12.0 to 4.13.0 (#10038)

Bumps [@typescript-eslint/types](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/visitor-keys) from 4.12.0 to 4.13.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/visitor-keys/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v4.13.0/packages/visitor-keys)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

fisker committed:
Enable `unicorn/prefer-regexp-test` (#10045)

dependabot[bot] committed:
Build(deps-dev): Bump webpack from 5.12.3 to 5.13.0 (#10040)

Bumps [webpack](https://github.com/webpack/webpack) from 5.12.3 to 5.13.0.
- [Release notes](https://github.com/webpack/webpack/releases)
- [Commits](https://github.com/webpack/webpack/compare/v5.12.3...v5.13.0)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dcyriller committed:
handlebars: Print final blockParam on its own line (#9978)

dependabot[bot] committed:
Build(deps-dev): Bump eslint-plugin-unicorn from 26.0.0 to 26.0.1 (#10056)

Bumps [eslint-plugin-unicorn](https://github.com/sindresorhus/eslint-plugin-unicorn) from 26.0.0 to 26.0.1.
- [Release notes](https://github.com/sindresorhus/eslint-plugin-unicorn/releases)
- [Commits](https://github.com/sindresorhus/eslint-plugin-unicorn/compare/v26.0.0...v26.0.1)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump webpack from 5.13.0 to 5.14.0 (#10054)

Bumps [webpack](https://github.com/webpack/webpack) from 5.13.0 to 5.14.0.
- [Release notes](https://github.com/webpack/webpack/releases)
- [Commits](https://github.com/webpack/webpack/compare/v5.13.0...v5.14.0)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump webpack from 5.13.0 to 5.14.0 in /website (#10053)

Bumps [webpack](https://github.com/webpack/webpack) from 5.13.0 to 5.14.0.
- [Release notes](https://github.com/webpack/webpack/releases)
- [Commits](https://github.com/webpack/webpack/compare/v5.13.0...v5.14.0)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps): Bump @angular/compiler from 11.0.8 to 11.0.9 (#10055)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

sosukesuzuki committed:
Add meriyah to getMarkdownSyntax (#10061)

dependabot[bot] committed:
Build(deps): Bump meriyah from 3.1.6 to 4.0.0 (#10060)

* Build(deps): Bump meriyah from 3.1.6 to 4.0.0

Bumps [meriyah](https://github.com/meriyah/meriyah) from 3.1.6 to 4.0.0.
- [Release notes](https://github.com/meriyah/meriyah/releases)
- [Changelog](https://github.com/meriyah/meriyah/blob/master/CHANGELOG.md)
- [Commits](https://github.com/meriyah/meriyah/compare/v3.1.6...v4.0.0)

Signed-off-by: dependabot[bot] <support@github.com>

* `FieldDefinition` -> `PropertyDefinition`

* `PrivateName` -> `PrivateIdentifier`

* Test https://github.com/meriyah/meriyah/issues/164

* Copy test from https://github.com/meriyah/meriyah/commit/f21882c312284572ac6d7e7630c4a677d6caed92

* Revert breaking change from 3cdb2a4edc8af0b1c02b1d3f1e1493550164fed0

* Fix test options

* Typo

* Update snapshot

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: fisker <lionkay@gmail.com>

sosukesuzuki committed:
Refactor: Use getLastStatement function in printStatementSequence (#10066)

* Add getLastStatement function

* Use loop

* Fix lint problem

sosukesuzuki committed:
Rename shouldNotPrePrintHardline to shouldPrePrintHardline (#10067)

dependabot[bot] committed:
Build(deps-dev): Bump eslint from 7.17.0 to 7.18.0 (#10070)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump webpack from 5.14.0 to 5.15.0 in /website (#10072)

dependabot[bot] committed:
Build(deps-dev): Bump @rollup/plugin-node-resolve from 11.0.1 to 11.1.0 (#10069)

thorn0 committed:
Disable Babel's error recovery for omitted call arguments (#10065)

sosukesuzuki committed:
Remove workaround for rollup/plugin-node-resolve 11 (#10075)

dependabot[bot] committed:
Build(deps-dev): Bump webpack from 5.14.0 to 5.15.0 (#10074)

dependabot[bot] committed:
Build(deps-dev): Bump rollup from 2.36.1 to 2.36.2 (#10071)

dcyriller committed:
handlebars: Support `htmlWhitespaceSensitivity=strict` option (2/3) (#9885)

* hbs: Add a strict mode

* Fix lint (after concat removal)

* Another lint error

thorn0 committed:
Smarter formatting of HTML class attribute (#7865)

* html: break class attr onto multiple lines, but try keeping consecutive classes with the same prefix on one line

* add changelog

* add more tests

* improve doc structure to fix an issue with the closing quote

* Update changelog_unreleased/html/pr-7865.md

Co-Authored-By: Jed Fox <git@twopointzero.us>

* handle classes with leading _ and -

* update snapshots

* Lint

* Update pr-7865.md

* Update changelog entry

Co-authored-by: Jed Fox <git@twopointzero.us>
Co-authored-by: fisker Cheung <lionkay@gmail.com>

thorn0 committed:
Add a test for HTML (#10077)

dependabot[bot] committed:
Build(deps-dev): Bump js-yaml from 3.14.1 to 4.0.0 in /website (#9996)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: fisker <lionkay@gmail.com>

dependabot[bot] committed:
Build(deps-dev): Bump eslint-config-prettier from 7.1.0 to 7.2.0 (#10082)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps): Bump @typescript-eslint/typescript-estree (#10084)

thorn0 committed:
docs: add a note on formatting reversibility (#10062)

dcyriller committed:
Document --htmlWhitespaceSensitivity option for Handlebars (#10079)

* Document htmlWhitespaceSensitivity for Handlebars

* Update options.md

Co-authored-by: Georgii Dolzhykov <thorn.mailbox@gmail.com>

thorn0 committed:
copy rationale.md update to stable docs

sosukesuzuki committed:
Improve visual separation between header and body in classes with multiline headers (#10085)

* Add hardline for a class whose extends part is broken

refactor

Use groupId

Remove isPreviousLineEmpty

* Avoid leading comments

* Rename to heritage

* Add heuristic for implements

* Fix lint problems

* Fix

Fix

Fix

* Extract getGroupId patterns

* Adopots solution 2

* Fix

* Fix lint problem

* Add changelog

* Address reviews

* Update tests

* Rename to createGroupIdMapper

* Split files

* Add tests for complex extends

* Fix changelog

dependabot[bot] committed:
Build(deps-dev): Bump rollup from 2.36.2 to 2.37.0 (#10093)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump webpack from 5.15.0 to 5.16.0 (#10091)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

thorn0 committed:
Fix webpack 5 CLI args in `yarn start` in website

dependabot[bot] committed:
Build(deps-dev): Bump rollup from 2.37.0 to 2.37.1 (#10099)

Bumps [rollup](https://github.com/rollup/rollup) from 2.37.0 to 2.37.1.
- [Release notes](https://github.com/rollup/rollup/releases)
- [Changelog](https://github.com/rollup/rollup/blob/master/CHANGELOG.md)
- [Commits](https://github.com/rollup/rollup/compare/v2.37.0...v2.37.1)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump @typescript-eslint/types from 4.13.0 to 4.14.0 (#10083)

Bumps [@typescript-eslint/types](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/visitor-keys) from 4.13.0 to 4.14.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/visitor-keys/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v4.14.0/packages/visitor-keys)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps-dev): Bump @glimmer/reference from 0.73.1 to 0.73.2 (#10104)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

fisker committed:
Update `eslint-plugin-unicorn` to v27 (#10110)
